analysis/monkmistweaver/src/modules/core/HotTrackerMW.tsx

```diff
@@ -23,37 +23,39 @@ class HotTrackerMW extends HotTracker {
     this.upwellingActive = selectedCombatant.hasTalent(SPELLS.UPWELLING_TALENT.id);
   }
 
-  calculateMaxDuration(event: ApplyBuffEvent & { selectedCombatant: Combatant }) {
+  calculateMaxDuration(event: ApplyBuffEvent, combatant: Combatant) {
     const spellId = event.ability.guid;
     if (spellId === SPELLS.ENVELOPING_MIST.id) {
-      return (ENV_BASE_DURATION + (this.mistwrapActive ? MISTWRAP : 0)) * 2;
+      return (
+        (ENV_BASE_DURATION + (combatant.hasTalent(SPELLS.MIST_WRAP_TALENT.id) ? MISTWRAP : 0)) * 2
+      );
     }
     if (spellId === SPELLS.ESSENCE_FONT_BUFF.id) {
-      return (EF_BASE_DURATION + (this.upwellingActive ? UPWELLING : 0)) * 2;
+      return (
+        (EF_BASE_DURATION + (combatant.hasTalent(SPELLS.UPWELLING_TALENT.id) ? UPWELLING : 0)) * 2
+      );
     }
     if (
-      event.selectedCombatant.hasBuff(SPELLS.THUNDER_FOCUS_TEA.id) &&
+      combatant.hasBuff(SPELLS.THUNDER_FOCUS_TEA.id) &&
       spellId === SPELLS.RENEWING_MIST_HEAL.id
     ) {
       return (REM_BASE_DURATION + TFT_REM_EXTRA_DURATION) * 2;
     }
     return REM_BASE_DURATION * 2;
   }
 
-  calculateMaxRemDuration(
-    event: (ApplyBuffEvent | RefreshBuffEvent) & { selectedCombatant: Combatant },
-  ) {
+  calculateMaxRemDuration(event: ApplyBuffEvent | RefreshBuffEvent, combatant: Combatant) {
```

Comment:
tacking an extra field onto the existing passed Event made no sense and is very awkward in TypeScript. I instead just made the combatant an extra parameter.